integration/yarn/src/main/java/alluxio/yarn/Client.java

```diff
@@ -367,4 +372,42 @@ private void setupApplicationSubmissionContext() {
     // Set the priority for the application master
     mAppContext.setPriority(Priority.newInstance(mAmPriority));
   }
+
+  /**
+   * Monitors the submitted application util app is running, finished, killed or failed.
+   *
+   * @throws YarnException if errors occur when obtaining application report from ResourceManager
+   * @throws IOException if errors occur when obtaining application report from ResourceManager
+   */
+  private void monitorApplication() throws YarnException, IOException {
+    while (true) {
+      // Check app status every 10 second.
+      CommonUtils.sleepMs(10 * Constants.SECOND_MS);
+      // Get application report for the appId we are interested in
+      ApplicationReport report = mYarnClient.getApplicationReport(mAppId);
+
+      YarnApplicationState state = report.getYarnApplicationState();
+      FinalApplicationStatus dsStatus = report.getFinalApplicationStatus();
+      switch (state) {
+        case RUNNING:
+          System.out.println("Application is running. Tracking url is " + report.getTrackingUrl());
+          return;
+        case FINISHED:
+          if (FinalApplicationStatus.SUCCEEDED == dsStatus) {
+            System.out.println("Application has completed successfully");
+          } else {
+            System.out.println("Application did finished unsuccessfully. YarnState="
+                + state.toString() + ", DSFinalStatus=" + dsStatus.toString());
+          }
+          return;
+        case KILLED:
+        case FAILED:
+          System.out.println("Application did not finish. YarnState=" + state.toString()
+              + ", DSFinalStatus=" + dsStatus.toString());
+          return;
+        default:
+          System.out.println("Application is " + state + ". Waiting.");
```

Comment:
maybe change this to ("Application is in state " + state + ". Waiting.") since "Application is in NEW" or "Application is NEW_SAVING" might be confusing